webdav/webdav_test.go

```diff
@@ -347,3 +370,64 @@ func TestFilenameEscape(t *testing.T) {
 		}
 	}
 }
+
+func TestMoveLockedSrcUnlockedDst(t *testing.T) {
+	// This test reproduces https://github.com/golang/go/issues/43556
+	do := func(method, urlStr string, body string, wantStatusCode int, headers ...string) (http.Header, error) {
```

Comment:
I think these tests could do with some refactoring. I tend to favour writing simple test harnesses that handle much of the common behaviour between tests. In this case, I think we could refactor the 'do' function as well as the server setup. That would simplify both the existing tests, and any new ones which use a similar pattern. I was going to do it in this PR, but it probably makes more sense to do it in a separate one specific to that refactoring. Let me know if you're interested.